platform-modules/mimetype-manager/src/main/scala/org/sunbird/mimetype/mgr/impl/CollectionMimeTypeMgrImpl.scala

```diff
@@ -34,6 +34,10 @@ class CollectionMimeTypeMgrImpl(implicit ss: StorageService) extends BaseMimeTyp
 		validate(node).map(res => getEnrichedMetadata(node.getMetadata.getOrDefault("status", "").asInstanceOf[String]))
 	}
 
+	override def publish(objectId: String, node: Node)(implicit ec: ExecutionContext, ontologyEngineContext: OntologyEngineContext): Future[Map[String, AnyRef]] = {
```

Comment:
The common method should be in the base call. Each MimetypeMarImple will override this method if there are any changes/modifications required.

This will avoid a lot of duplication code in all the MimetypeMarImple

Reply:
That is how it is implemented. Base method is in 'platform-modules/mimetype-manager/src/main/scala/org/sunbird/mimetype/mgr/MimeTypeManager.scala'. This method is being overriden in respective mimeType managers. Hence, you see the keyword '**override**' at the start of the method declaration.

Reply:
You can also refer to the review flow. Same coding pattern has been followed.